### PR TITLE
Serialize dynamo/inductor config for minifier

### LINKED ISSUE #1965: Preserve dynamo/inductor config during minifier
Dynamo/inductor config is not preserved, so minifier might run with different settings - e.g. this might happen if a user changes config at runtime. This is especially relevant with the new `torch.compile` inductor modes, since inductor configs are changed at runtime.

### PR DESCRIPTION
**Pull request by williamwen42**
Fixes https://github.com/pytorch/torchdynamo/issues/1965

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_dynamo/config_utils.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import pickle
+
+
+# Construct functions that save/load the state of the config module `module`.
+# The config settings are expected to either be module-level globals or
+# class variables.
+# `ignore_set` is a set of names of configurations to ignore. e.g. if you
+# want to ignore config.x and config.y.z in your config module, then
+# `ignore_set` should be {"x", "y.z"}.
+def get_config_serialization_fns(module, ignore_set=None):
```
Comment (mlazos) [inline torch/_dynamo/config_utils.py:11]: Shouldn't we just override `__getstate__` and `__setstate__` for the `_AccessLimitingConfig`

**Review by mlazos — CHANGES_REQUESTED** [review]

File: torch/_dynamo/config_utils.py (hunk 1 above)
Comment (williamwen42) [reply torch/_dynamo/config_utils.py:11]: Inductor config doesn't use a class such as _AccessLimitingConfig.

**Review by mlazos — APPROVED** [review]

**Pull request closed without merging**